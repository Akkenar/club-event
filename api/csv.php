<?php

$value_separator = "\t";
$line_separator = "\n";

function getHeaderLineFromQuery($records)
{
  $header = '';
  global $value_separator;

  $fields = fieldNames($records);
  for ($i = 0; $i < count($fields); $i++) {
    $header .= fieldName($records, $i) . $value_separator;
  }

  return $header;
}

function getBodyLinesFromQuery($records)
{
  $data = '';
  global $value_separator;
  global $line_separator;

  foreach ($records->fetch_all() as $row) {
    $line = '';
    foreach ($row as $value) {
      if (!isset($value) || $value == "") {
        $value = $value_separator;
      } else {
        $value = str_replace('"', '""', $value);
        $value = '"' . $value . '"' . $value_separator;
      }
      $line .= $value;
    }
    $data .= trim($line) . $line_separator;
  }
  return str_replace("\r", "", $data);
}
