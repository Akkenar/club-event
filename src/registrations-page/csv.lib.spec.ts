import mockRegistrations from '../../mocks/registrations.json';
import { convertToCsv, exportToCsv } from './csv.lib';

describe('csv', () => {
  it('should convert all registrations to CSV', () => {
    const result = convertToCsv(mockRegistrations as any);

    // Because why not using snapshots to test string data!
    expect(result).toMatchSnapshot();
  });

  it('should download the data', () => {
    const createElementSpy = jest.spyOn(document, 'createElement');
    exportToCsv(mockRegistrations as any);
    expect(createElementSpy).toHaveBeenCalledWith('a');
  });
});
