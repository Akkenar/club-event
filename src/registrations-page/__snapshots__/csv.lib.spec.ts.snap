// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csv should convert all registrations to CSV 1`] = `
"Identifiant technique	Date	Référence	Langue	Prénom	Nom	E-mail	Club	Address	Numéro	NPA	Localité	A payer	Nombre de types de commande	Petit Déjeuner	Souper	Souper	Pique-nique	Nuit PC	Nuit Camping
1	2018-12-28 14:21:04	ROO-5C26315	fr	Tia	Roob	pierredominique.putallaz@gmail.com	GSL	Maggie Haven	5A	1018	Barrowshaven	65	1		1				
2	2018-12-28 14:29:48	KON-5C26335	fr	Cayla	Konopelski	pierredominique.putallaz@gmail.com	GSL	Parker Expressway	5A	1018	Port Margueritemouth	1080	6	4	6	4	2	6	6
3	2018-12-28 14:30:19	STR-5C26337	de	Paige	Streich	pierredominique.putallaz@gmail.com	GSL	Randi Knolls	5A	1018	Nolanside	745	5		0	3	5	2	8"
`;
